### ISSUE TITLE
Wrong page sizes returned by `pdf-view-desired-image-size` when `pdf-view-auto-slice-minor-mode` is enabled

### ISSUE THREAD

**Issue by nnicandro** [politza/pdf-tools#474]
TL;DR

- Consider the bounding box of the page argument passed to `pdf-view-desired-image-size` instead of the slice of the current page when `pdf-view-auto-slice-minor-mode` is enabled.

I've been trying to figure out why I had previously thought that the page
sizes can vary by a wide margin across different PDFs (see #467). When I went back and
tried to look into this further for some of my PDFs, I found that the image sizes
calculated via `pdf-view-desired-image-size` weren't so different. At least not
so different to account for 100 MB differences in memory usage
across PDFs that I though I was observing.

While investigating this, I found an issue with `pdf-view-desired-image-size`.
In particular, its use of the current slice to calculate
sizes and the consequences this has when `pdf-view-auto-slice-minor-mode` is
enabled.

<https://github.com/politza/pdf-tools/blob/bcb1fa4597feb5a9d44983444054c9ca68f5ca0f/lisp/pdf-view.el#L1099>

Suppose we have `pdf-view-display-size` set to `fit-width` (the default), with
`pdf-view-auto-slice-minor-mode` enabled. Then, since `pdf-view-current-slice`
is used to calculate sizes, `pdf-view-desired-image-size` returns the wrong
size for pages that are not the current page. So it looks like the
bounding box of the page passed to `pdf-view-desired-image-size` should be considered.
At least when `pdf-view-auto-slice-minor-mode` is enabled.

This has consequences for the page pre-fetcher as well since
`pdf-cache--prefetch-start` calls `pdf-view-desired-image-size`. If the current
page has a small bounding box, then the scale used in the size calculation will
be large and the sizes of all pages pre-fetched whenever that page is current
will be large.

Below is some code which shows the problem:

-   Open `pdf-tools/test/test.pdf`
-   Run the following function in the PDF buffer

```emacs-lisp
    (defun pdf-view-test-desired-image-size ()
      (pdf-view-auto-slice-minor-mode 1)
      (pdf-view-fit-width-to-window)
      (pdf-view-goto-page 1)
      (let ((sizes (mapcar #'pdf-view-desired-image-size '(1 2 3))))
        (pdf-view-goto-page 2)
        (message "sizes on page 1 %s" sizes)
        (message "sizes on page 2 %s"
    	     (mapcar #'pdf-view-desired-image-size '(1 2 3)))))
```

I get

```
    sizes on page 1 ((2047 . 2649) (2047 . 2649) (2047 . 2649))
    sizes on page 2 ((4800 . 7058.0) (4800 . 7058.0) (4800 . 7058.0))
```

**Comment by politza** [comment]
I don't see why the regular (i.e. non-prefetched) images should have the
wrong size, since the auto-slice mode adjusts the slice before the image
is rendered and displayed.

Though, I agree that the prefetcher has a problem here.

**Comment by dzop** [comment]
Yes your right, there isn't an issue with the non-prefetched images.

Can't we just extract out the code from `pdf-view-set-slice-from-bounding-box` into a function `pdf-view-slice-from-bounding-box` and just use that for the slice in `pdf-view-desired-image-size`? We would probably still have to clamp the numbers in the slice like what is done in `pdf-view-set-slice` or would we? Doesn't epdfinfo already clamp those numbers?
